### PR TITLE
fix(amplify_analytics_pinpoint): fix plugin interface naming

### PR DESCRIPTION
**Pull request by fjnoyp**
analytics_plugin_interface —> amplify_analytics_plugin_interface

To ensure consistent naming between file name and class name.

**NOTE **

This PR is incomplete!  The pubspec.yaml must be updated in amplify_analytics_pinpoint to use the new version of amplify_analytics_plugin_interface with this change or else it won't work.  Yet you'll need to publish these changes first to make that happen! .... 

You cannot run dart unit tests and the pubspec.yaml in amplify_anlytics_pinpoint still relies on the old amplify_analytics_plugin_interface 

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by jamesonwilliams — APPROVED** [review]

**Comment by haverchuck** [comment]
@fjnoyp Why can't we roll the additional pubspec changes into this PR and then release them sequentially?

**Comment by fjnoyp** [comment]
Yeah we should have the pubspec changes in this PR before we merge it.  I'm just not sure which version to use @haverchuck

**Comment by haverchuck** [comment]
@fjnoyp Ok, I'll approve this PR for now.

**Review by haverchuck — APPROVED** [review]

**Comment by Amplifiyer** [comment]
This requires some more changes in amplify_flutter package to update the class name

File: packages/amplify_analytics_plugin_interface/lib/src/Analytics/AnalyticsUserProfile.dart [hunk 1]
```diff
@@ -13,19 +13,21 @@
...
   String email;
   String plan;
   AnalyticsUserProfileLocation location;
   AnalyticsProperties properties;
 
-  AnalyticsUserProfile({
-    String name, String email, String plan, AnalyticsUserProfileLocation location, AnalyticsProperties analyticsProperties
-  }) {
+  AnalyticsUserProfile(
+      {String name,
```
Comment (jamesonwilliams) [inline packages/amplify_analytics_plugin_interface/lib/src/Analytics/AnalyticsUserProfile.dart:26]: A trivial suggestion to keep the params at the same column!
```suggestion
  AnalyticsUserProfile({
      String name,
```

**Review by jamesonwilliams — APPROVED** [review]

**Review by Amplifiyer — APPROVED** [review]
🚀

**Pull request merged**